feat(index.js, RSVPModel.js, rsvp.js): add rsvp model and routes

michaeltran3599 opened the pull request:
add rsvp model and routes to create and get an authenticated user's rsvp

michaeltran3599 committed:
feat(index.js, RSVPModel.js, rsvp.js): add rsvp model and routes

add rsvp model and routes to create and get an authenticated user's rsvp

michaeltran3599 committed:
style(rsvp.js, RSVPModel.js): fixed lint errors

Jumer-Caragay commented on src/models/RSVPModel.js:
```diff
@@ -0,0 +1,57 @@
+    min: 0,
+    max: 4,
+    required: function () { return this.attending }
+
```
Remove Blank Line

Jumer-Caragay commented on src/models/RSVPModel.js:
```diff
@@ -0,0 +1,57 @@
+    min: 0,
+    max: 2,
+    required: function () { return this.attending && this.bus === 1 }
+
```
Remove additional blank line.

Jumer-Caragay commented on src/models/RSVPModel.js:
```diff
@@ -0,0 +1,57 @@
+    required: function () { return this.attending }
+
+  },
+  travelType: {
```
This tag should be transportation check out https://github.com/slohacks/attendee-application/blob/master/src/containers/data/rsvpData.json

Jumer-Caragay commented on src/models/RSVPModel.js:
```diff
@@ -0,0 +1,57 @@
+    max: 2,
+    required: function () { return this.attending }
+  },
+  bus: {
```
should be "buses"

Jumer-Caragay commented on src/models/RSVPModel.js:
```diff
@@ -0,0 +1,57 @@
+    required: function () { return this.attending && this.travelType === 0 }
+  },
+  socal: {
+    type: Number,
```
Currently, the frontend sends this as a string. thoughts on number vs string?

cfspecht replied:
Tommy and I think it is easier to store this data as a string as well, I am updating the Application model to reflect this.

Jumer-Caragay commented on src/models/RSVPModel.js:
```diff
@@ -0,0 +1,57 @@
+    max: 2,
+    required: function () { return this.attending && this.bus === 0 }
+  },
+  norcal: {
```
Same comment as above.

Jumer-Caragay commented on src/models/RSVPModel.js:
```diff
@@ -0,0 +1,57 @@
+const mongoose = require('mongoose')
+
+const RSVP = mongoose.model('RSVP', {
```
Think about how we should expire out RSVP, ex if user is accepted but RSVPS after 7 days how do we prevent them from creating a document

Jumer-Caragay commented on src/routers/rsvp.js:
```diff
@@ -0,0 +1,33 @@
+  }
+})
+
+router.get('/rsvp/', authMiddleware, async (req, res) => {
```
/rsvp/:id is more concise, Change it to that implementation. I remember talking about it doesnt make sense for us to pull a specific RSVP but just to be consistent with other routes/models, change the implementation to reflect that.

Jumer-Caragay reviewed (changes requested):
If you have any questions about the comments, feel free to ask!

In addition to the conditional required and field types of the model,
consider looking at the implementation of the frontend
https://github.com/slohacks/attendee-application/blob/master/src/containers/RSVP.jsx
I'll look more into this.

Another implementation is ignore conditional required fields 
and just initiate them to null.

michaeltran3599 committed:
feat(RSVPModel.js, rsvp.js, authMiddleware.js, index.js): Add slohacks user admin rights

Added email check in authMiddleware to give @slohacks.com users access to all rsvps

cfspecht reviewed (changes requested):
Talk with Tommy about storing the data as integers vs. strings, and Jumer's changes. Looks good otherwise.
